### PR TITLE
chore(docs): Add documentation for tailwindcss support

### PR DESCRIPTION
**Pull request by nextlevelshit**
# Description

I was missing a documentation how to implement the tailwindcss framework. After searching the web I found lot of blog posts about how to implement tailwindcss with postcss plugin, but not with sass/scss support.

This plugin looks to be perfect for this and makes it so easy to implement tailwindcss, but there is somethings to care for. That's why I tought about adding a brief instruction how to implement tailwindcss.

Additionally I've found some mistakes in the highlighting and some missing information about which file is displayed in the code samples. Instead of splitting it into different PR's I've done it at once to save some time.

### CODE REVIEW

File: packages/gatsby-plugin-sass/README.md [hunk 1]
```diff
@@ -152,6 +154,29 @@ and common cross-browser flexbox bugs. Normally you don't need to think about it
 you'd prefer to add additional postprocessing to your Sass output you can specify plugins
 in the plugin options.
 
+#### TailwindCSS
+
+To be able to use [TailwindCSS](https://tailwindcss.com/) within your Sass/SCSS files, add the `tailwindcss` package into the `postCSSPlugins` parameter.
```
Comment (marcysutton) [inline packages/gatsby-plugin-sass/README.md]: We should mention how to install Tailwind, as well.

File: packages/gatsby-plugin-sass/README.md [hunk 2]
```diff
@@ -26,15 +27,16 @@ html {
...
+import("./src/index.scss")
 ```
 
 ## Other options
 
 If you need to pass options to Sass use the plugins options, see [node-sass](https://github.com/sass/node-sass)/[dart-sass](https://github.com/sass/dart-sass) docs
 for all available options.
 
-```javascript
+```js
```
Comment (marcysutton) [inline packages/gatsby-plugin-sass/README.md]: Both `js` and `javascript` are used throughout the docs, and your later entry uses `javascript`...perhaps we should make them consistent, at least in the same file?

File: packages/gatsby-plugin-sass/README.md [hunk 3]
```diff
@@ -152,6 +154,30 @@ and common cross-browser flexbox bugs. Normally you don't need to think about it
...
+      postCssPlugins: [
+        require('tailwindcss'),
+        // require('./tailwind.config.js') // Load custom tailwindcss configuration 
+      ]
+    },
+  }
+]
+```
+
+Learn here [how to customize your tailwindcss configuration](https://tailwindcss.com/docs/configuration#creating-your-configuration-file).
```
[the remark this answers is not in the document]
Comment (nextlevelshit) [reply packages/gatsby-plugin-sass/README.md]: Did this already ;) https://github.com/tailwindcss/docs/pull/287

**Comment by nextlevelshit** [comment]
> This is awesome!
> 
> Can you also add some info to the [TailwindCSS docs page](https://www.gatsbyjs.org/docs/tailwind-css/)? We could link to it from the [Sass page](https://www.gatsbyjs.org/docs/sass/), too.

Already done: https://github.com/tailwindcss/docs/pull/287/files#diff-c3d53ed7768a8c093507e9455ef030d2R304

File: packages/gatsby-plugin-sass/README.md [hunk 4]
```diff
@@ -160,10 +186,10 @@ Using [resolve-url-loader](https://github.com/bholloway/resolve-url-loader) prov
 
 First:
 
-```javascript
-  npm install resolve-url-loader --save-dev
-  or
-  yarn add resolve-url-loader --dev
+```bash
+npm install resolve-url-loader --save-dev
+or
```
[the remark this answers is not in the document]
Comment (sagarhani) [reply packages/gatsby-plugin-sass/README.md]: Hey,
This should be `shell`.
https://github.com/gatsbyjs/gatsby/pull/17694

**Comment by LekoArts** [comment]
I personally think that this shouldn't go into the sass plugin docs and rather into the already existing Tailwind docs: https://www.gatsbyjs.org/docs/tailwind-css/

It could be `Option #3` there and only would be linked to from the sass docs.

**Comment by nextlevelshit** [comment]
> I personally think that this shouldn't go into the sass plugin docs and rather into the already existing Tailwind docs: https://www.gatsbyjs.org/docs/tailwind-css/
> 
> It could be `Option #3` there and only would be linked to from the sass docs.

Very good idea! Thanks for the hint. IMHO this PR can be closed now.

**Comment by LekoArts** [comment]
Some of your changes to the code blocks are still valid, so we'd like to keep that and you can of course change your PR to reflect what I proposed. However, please wait for Marcy's answer first before beginning to change things. Thanks!

**Comment by nextlevelshit** [comment]
> Some of your changes to the code blocks are still valid, so we'd like to keep that and you can of course change your PR to reflect what I proposed. However, please wait for Marcy's answer first before beginning to change things. Thanks!

Sorry, I was confused. I will wait. Thanks.

**Comment by marcysutton** [comment]
That's a good point @LekoArts, sounds like a good approach to me! Thanks for the input 🌟

**Comment by nextlevelshit** [comment]
To not get it wrong @LekoArts and @marcysutton :

1. Remove TailwindCSS instructions from  `packages/gatsby-plugin-sass/README.md`
2. But keep that little code block improvements
3. Add TailwindCSS instructions to `docs/docs/tailwind-css.md`

I am not that familiar with your workflows. So, I hope I've got it right finally. Thanks for your support.

**Comment by LekoArts** [comment]
Yeah, sounds right! Thanks :)

**Comment by nextlevelshit** [comment]
@marcysutton I've done my changes and added it to my `patch-3` branch, but to be honest, shouldn't I rebae my branch to the current master? I mean the sense of git is to manage previous changes with mine, but for the sake of consistence and decreasing the probability of distraction through merge requests it might be good to rebase my branch befor merging. What do you think?

**Comment by nextlevelshit** [comment]
@marcysutton I've done my changes and added it to my patch-3 branch, but to be honest, shouldn't I rebae my branch to the current master? I mean the sense of git is to manage previous changes with mine, but for the sake of consistence and decreasing the probability of distraction through merge requests it might be good to rebase my branch before merging. What do you think?

File: packages/gatsby-plugin-sass/README.md [hunk 5]
```diff
@@ -25,17 +25,16 @@ html {
...
 ```
 
 ## Other options
 
 If you need to pass options to Sass use the plugins options, see [node-sass](https://github.com/sass/node-sass)/[dart-sass](https://github.com/sass/dart-sass) docs
 for all available options.
 
-```javascript
-// in gatsby-config.js
+```javascript:title="gatsby-config.js"
```
Comment (LekoArts) [inline packages/gatsby-plugin-sass/README.md]:
```suggestion
```javascript:title=gatsby-config.js
```

Please without the `""`. Also applies to the titles below

File: packages/gatsby-plugin-sass/README.md [hunk 6]
```diff
@@ -104,12 +100,11 @@ plugins: [
 ]
 ```
 
-### Bootstrap 3 (with `bootstrap-sass`)
+#### Bootstrap 3 (with `bootstrap-sass`)
```
Comment (LekoArts) [inline packages/gatsby-plugin-sass/README.md]:
```suggestion
### Bootstrap 3 (with `bootstrap-sass`)
```

Please keep the correct heading levels

File: packages/gatsby-plugin-sass/README.md [hunk 7]
```diff
@@ -160,15 +154,15 @@ Using [resolve-url-loader](https://github.com/bholloway/resolve-url-loader) prov
 
 First:
 
-```javascript
-  npm install resolve-url-loader --save-dev
-  or
-  yarn add resolve-url-loader --dev
+```bash
```
Comment (LekoArts) [inline packages/gatsby-plugin-sass/README.md]:
```suggestion
```shell
```

Please also remove the yarn instructions and only keep "npm"

File: docs/docs/tailwind-css.md [hunk 8]
```diff
@@ -105,6 +106,41 @@ const Button = tw.button`
...
 
+### Option #3: Sass/SCSS
+
+1.  Install the Gatsby Sass plugin [**gatsby-plugin-sass**](https://www.gatsbyjs.org/packages/gatsby-plugin-sass/) and `node-sass`.
+
+```shell
+npm install --save node-sass gatsby-plugin-sass
+```
+
+_The default Sass implementation wil lbe `node-sass`, but can be changed. Alternatively you can switch to the Dart implementation `sass`:_
```
Comment (LekoArts) [inline docs/docs/tailwind-css.md]: I don't think it's necessary to overcomplicate this guide here and also explain the Dart implementation. That's already covered in the README of the SASS plugin. Let's provide a happy path with `node-sass` and leave a note that for the Dart implementation people should look into the README

Comment (nextlevelshit) [reply docs/docs/tailwind-css.md]: There's a little typo: "wil lbe" > "will be"

Comment (pieh) [reply docs/docs/tailwind-css.md]: Agree with @LekoArts - let's remove lines 117-223 as this just distraction

**Comment by pieh** [comment]
Hey @nextlevelshit! There are still few not addressed inline comments from @LekoArts

**Comment by LekoArts** [comment]
I'll fix the issues today and merge.

**Pull request merged**